gui/src/installer/view.rs

```diff
@@ -937,26 +937,58 @@ pub fn install<'a>(
                                                     acc.push(
                                                         Row::new()
                                                             .spacing(5)
-                                                            .push(text(hw.0.to_string()).small())
                                                             .push(
-                                                                text(format!(
-                                                                    "(fingerprint: {})",
-                                                                    hw.1
-                                                                ))
+                                                                text(
+                                                                    context
+                                                                        .keys
+                                                                        .iter()
+                                                                        .find_map(|k| {
+                                                                            if k.master_fingerprint
+                                                                                == hw.1
+                                                                            {
+                                                                                Some(k.name.clone())
+                                                                            } else {
+                                                                                None
+                                                                            }
+                                                                        })
+                                                                        .unwrap(),
```

Comment:
We should only ever crash if continuing would make the application in an inconsistent state. It's not the case here, we could just not show the alias.

Actually i managed to trigger this assertion (albeit in a convoluted manner but still):
```
2023-04-18T09:27:23.113371Z ERROR liana_gui:417: panic occurred at line 954 of file src/installer/view.rs: Some("called `Option::unwrap()` on a `None` value")                                                                                                                                                                                                           
   0: liana_gui::setup_panic_hook::{{closure}}                                                                                                                                                                                                                                                                                                                             
             at src/main.rs:411:18                                                                                                                                                                                                                                                                                                                                         
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call                                                                                                                                                                                                                                                                                                      
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:2002:9                                                                                                                                                                                                                                                                          
      std::panicking::rust_panic_with_hook                                                                                                                                                                                                                                                                                                                                 
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:692:13                                                                                                                                                                                                                                                                        
   2: std::panicking::begin_panic_handler::{{closure}}                                                                                                                                                                                                                                                                                                                     
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:577:13                                                                                  
   3: std::sys_common::backtrace::__rust_end_short_backtrace                                                                                                                         
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:137:18                                                                       
   4: rust_begin_unwind                                                                                                                                                              
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:575:5                                                                                   
   5: core::panicking::panic_fmt                                                                                                                                                     
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:64:14                                                                                  
   6: core::panicking::panic                                                                                                                                                         
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:114:5                                                                                  
   7: core::option::Option<T>::unwrap                                                                                                                                                
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/option.rs:823:21                                                                                    
   8: liana_gui::installer::view::install::{{closure}}                                                                                                                               
             at src/installer/view.rs:942:69                                                                                                                                         
   9: core::iter::traits::iterator::Iterator::fold                                                                                                                                   
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/iter/traits/iterator.rs:2438:21                                                                     
  10: liana_gui::installer::view::install                                                                                                                                            
             at src/installer/view.rs:934:50                                                                                                                                         
  11: <liana_gui::installer::step::Final as liana_gui::installer::step::Step>::view                                                                                                  
             at src/installer/step/mod.rs:272:9                                                                                                                                      
  12: liana_gui::installer::Installer::view                                                                                                                                          
             at src/installer/mod.rs:191:9                                                                                                                                           
  13: <liana_gui::GUI as iced::application::Application>::view                                                                                                                       
             at src/main.rs:273:36                                                                                                                                                   
  14: <iced::application::Instance<A> as iced_native::program::Program>::view                                                                                                        
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced-0.7.0/src/application.rs:231:9                                                                   
  15: iced_winit::application::build_user_interface                                                                                                                                  
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_winit-0.7.0/src/application.rs:606:16                                                            
  16: iced_glutin::application::run_instance::{{closure}}                                                                                                                            
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_glutin-0.6.0/src/application.rs:327:43                                                           
  17: iced_glutin::application::run::{{closure}}                                                                                                                                     
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_glutin-0.6.0/src/application.rs:187:24                                                           
  18: winit::platform_impl::platform::sticky_exit_callback                                                                                                                           
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:849:9                                                     
  19: winit::platform_impl::platform::wayland::event_loop::EventLoop<T>::run_return                                                                                                  
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/wayland/event_loop/mod.rs:488:13                                 
  20: winit::platform_impl::platform::EventLoop<T>::run_return                                                                                                                       
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:748:56                                                    
  21: <winit::event_loop::EventLoop<T> as winit::platform::run_return::EventLoopExtRunReturn>::run_return                                                                            
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform/run_return.rs:62:9
  22: iced_glutin::application::run                                                       
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_glutin-0.6.0/src/application.rs:162:13
  23: iced::application::Application::run                                                 
             at /home/darosior/.cargo/registry/src/github.com-1ecc6299db9ec823/iced-0.7.0/src/application.rs:209:12
  24: liana_gui::main                                                                     
             at src/main.rs:393:21                                                        
  25: core::ops::function::FnOnce::call_once                                              
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:250:5
  26: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:121:18
  27: std::rt::lang_start::{{closure}}                                                    
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:166:18
  28: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:287:13
      std::panicking::try::do_call                                                        
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:483:40
      std::panicking::try                                                                 
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:447:19
      std::panic::catch_unwind                                                            
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panic.rs:140:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:148:48
      std::panicking::try::do_call                                                        
             at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:483:40
```

Reply:
Hi @darosior, thanks for pointing that out. I've now replaced this `unwrap` call with `push_maybe`, which I think should achieve the required behaviour.